### PR TITLE
Set output length in mbedTLS and TinyCrypt crypto driver shims

### PR DESCRIPTION
**Pull request by lpereira**
See #4684 for details

### CODE REVIEW

**Comment by MaureenHelm** [comment]
@vakulgarg Will you please have a look at this?

File: drivers/crypto/crypto_mtls_shim.c [hunk 1]
```diff
@@ -87,6 +93,11 @@ static int mtls_ccm_decrypt_auth(struct cipher_ctx *ctx,
 		return -EINVAL;
 	}
 
+	/* Only tag_len bytes are compared in mbedtls_ccm_auth_decrypt()
+	 * if the decryption was successful.
+	 */
+	apkt->pkt->out_len = ctx->mode_params.ccm_info.tag_len;
```
Comment (vakulgarg) [inline drivers/crypto/crypto_mtls_shim.c]: Why only tag_len is returned? Shouldn't it be same as in CBC decrypt case?

File: drivers/crypto/crypto_tc_shim.c [hunk 2]
```diff
@@ -155,6 +169,12 @@ static int do_ccm_decrypt_auth(struct cipher_ctx *ctx,
 		return -EIO;
 	}
 
+	/* Only tag_len bytes is compared in
+	 * tc_ccm_decryption_verification() before TC_CRYPTO_SUCCESS
+	 * can be returned.
+	 */
+	op->out_len = ccm_param->tag_len;
+
```
Comment (vakulgarg) [inline drivers/crypto/crypto_tc_shim.c]: It is not clear to me why you return only the tag length.

File: drivers/crypto/crypto_mtls_shim.c (hunk 1 above)
Comment (lpereira) [reply drivers/crypto/crypto_mtls_shim.c]: There's a comment right above that explaining the decision.  Does it make sense?

File: drivers/crypto/crypto_tc_shim.c (hunk 2 above)
Comment (lpereira) [reply drivers/crypto/crypto_tc_shim.c]: Is the comment right before the assignment not clear? What should be returned here instead?

File: drivers/crypto/crypto_mtls_shim.c (hunk 1 above)
Comment (vakulgarg) [reply drivers/crypto/crypto_mtls_shim.c]: My point is that the mtls_ccm_decrypt_auth() function is expected to return the length of decrypted text in 'out_len'. So irrespective of 'tag_len' bytes getting compared in mbedtls_ccm_auth_decrypt(), it has to be length of decrypted text. Tag_len is constant value for a given cipher.

File: drivers/crypto/crypto_tc_shim.c (hunk 2 above)
Comment (vakulgarg) [reply drivers/crypto/crypto_tc_shim.c]: Same counter argument as in above comment.

**Comment by vakulgarg** [comment]
Looks ok.

**Comment by lpereira** [comment]
If it looks OK, could we get an approval and merge the patch?

**Comment by vakulgarg** [comment]
Yes

**Review by nashif — APPROVED** [review]

**Pull request merged**